### PR TITLE
Added code to bin/deploy.py - to delete and re create the licenses.csv file on every deploy.

### LINKED ISSUE #79: licenses.csv is not generated from json datafiles
It looks like licenses.csv was introduced in https://github.com/okfn/licenses/pull/46 but is manually maintained, allowing it to get out of sync as noticed in #78 

It seems that `bin/deploy.py` ought generate the CSV file.

### PR DESCRIPTION
**Pull request by arielferdman**
This is a method of keeping the licenses.csv in sync with license data.

Fixes #79

### CODE REVIEW

File: bin/deploy.py [hunk 1]
```diff
@@ -4,75 +4,93 @@
...
-        ]
+    ]
+
+    def create_licenses_csv_file(self):
+        print('Updating csv file...')
+        licenses = self.get_licenses()
+        field_names = ['id', 'domain_content', 'domain_data', 'domain_software', 'family', 'is_generic', 'maintainer',
+                       'od_conformance', 'osd_conformance', 'status', 'title', 'url', 'legacy_ids']
+        filename = 'licenses.csv'
+        with open(filename, 'wb') as csv_file:
```
Comment (mlinksva) [inline bin/deploy.py:65]: Why binary?

Also, when I run locally, I get CRs added, perhaps due to https://docs.python.org/3/library/csv.html#id3 ?

Comment (arielferdman) [reply bin/deploy.py:65]: This should fix the CRs that you see. 
If you are using Windows - i think it should also behave the same, But i currently have no quick way to test it. 

Regarding the good question:
First this is what the python 2 (which is what the project is using) - docs say about binary mode in csv:
https://docs.python.org/2.7/library/csv.html#csv.writer

> If csvfile is a file object, it must be opened with the ‘b’ flag on platforms where that makes a difference.

Which i interpret as: since it is not stated and i am not sure where it does make a difference - i should use it because somewhere its needed and where its not needed - it does not make a difference.

These two stack overflow answers clarified it to me even further:

https://stackoverflow.com/questions/1170214/python-2-csv-writer-produces-wrong-line-terminator-on-windows/1170297#1170297

https://stackoverflow.com/questions/31483253/when-to-open-file-in-binary-mode-b/31483829#31483829

So basically as i understand it - writing the csv file to disk in binary mode - if used on a windows os - prevents windows from doing this:  
1.`csv_module --->  \r\n`
2.`\r[\n] ---> windows (thinking that its a text file) ---> \r[\r\n]` 
and that spare \r is what will probably be seen on some text editors.

Hope this fixes and clarifies the issue :)

Comment (mlinksva) [reply bin/deploy.py:65]: Thanks it does!

**Comment by mlinksva** [comment]
@arielferdman thanks! I added sorting the output hopefully to make future diffs of the csv better. I'm baffled as to why `met-office-cp` and `ukcrown` are first, but close enough.

Feel free to make further PRs if you have other improvements. 😄
